### PR TITLE
fix(ruff): adapt to new json output format

### PR DESCRIPTION
**Pull request by lithammer**
The JSON output format for fixes has changed causing parsing error:

> Linting with ruff failed:
> Cannot read properties of undefined (reading 'row')

### CODE REVIEW

File: src/features/linters/ruff.ts [hunk 1]
```diff
@@ -58,11 +70,14 @@ export class Ruff extends BaseLinter {
   private fixToWorkspaceEdit(filename: string, fix: IRuffFix): WorkspaceEdit | null {
     if (!fix) return null;
 
+    // TODO: Support multiple edits.
+    const edit = fix.edits[0];
```
Comment (lithammer) [inline src/features/linters/ruff.ts:74]: Not sure what the best way to support this is. But at least this fixes the immediate parsing error.

**Comment by fannheyward** [comment]
https://github.com/charliermarsh/ruff/pull/3709

File: src/features/linters/ruff.ts (hunk 1 above)
Comment (lithammer) [reply src/features/linters/ruff.ts:74]: By the way, here's an example of a fix with multiple edits.

```python
# foo.py
import subprocess
subprocess.run([], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

```console
$ ruff --format json --select UP foo.py
[
  {
    "code": "UP022",
    "message": "Sending stdout and stderr to pipe is deprecated, use `capture_output`",
    "fix": {
      "message": "Replace with `capture_output` keyword argument",
      "edits": [
        {
          "content": "capture_output=True",
          "location": {
            "row": 3,
            "column": 19
          },
          "end_location": {
            "row": 3,
            "column": 41
          }
        },
        {
          "content": "",
          "location": {
            "row": 3,
            "column": 41
          },
          "end_location": {
            "row": 3,
            "column": 65
          }
        }
      ]
    },
    "location": {
      "row": 3,
      "column": 1
    },
    "end_location": {
      "row": 3,
      "column": 67
    },
    "filename": "/<redacted>/foo.py",
    "noqa_row": 3
  }
]
```

**Comment by fannheyward** [comment]
@lithammer sorry for the delay, I made fix in #918, support old and new format both, also supports the multiple edits. 

```py
# foo.py
import subprocess
subprocess.run([], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

to

```py
import subprocess
subprocess.run([], capture_output=True)
```

**Comment by lithammer** [comment]
Thanks! Looking back at this, I don't think my fix was correct either. I was using `fix.message` instead of `fix.edits[*].content` which would've resulted in something like this:


```python
subprocess.run([], Sending stdout and stderr to pipe is deprecated, use `capture_output`)
```

**Pull request closed without merging**